doc/man3/DEFINE_STACK_OF.pod

```diff
@@ -229,6 +229,13 @@ A failed search is indicated by a B<-1> return value.
 STACK_OF(), DEFINE_STACK_OF(), DEFINE_STACK_OF_CONST(), and
 DEFINE_SPECIAL_STACK_OF() are implemented as macros.
 
+It is not an error to call B<sk_I<TYPE>_num>(), B<sk_I<TYPE>_value>(),
+B<sk_I<TYPE>_free>(), B<sk_I<TYPE>_zero>(), B<sk_I<TYPE>_pop_free>(),
+B<sk_I<TYPE>_delete>(), B<sk_I<TYPE>_delete_ptr>(), B<sk_I<TYPE>_pop>(),
+B<sk_I<TYPE>_shift>(), B<sk_I<TYPE>_find>(), B<sk_I<TYPE>_find_ex>(),
+and B<sk_I<TYPE>_find_all>() on a NULL stack, empty stack, or with
+an invalid index. An error is not raised in these conditions.
```

Comment:
Well, though better than nothing, this is a very coarse statement.
Yet for all others better add that calling them is discouraged on a NULL stack (except for `sk_<TYPE>_free` and  `sk_<TYPE>_pop_free`), with an invalid index, or on an empty stack while at least one element is needed.

Reply:
And `sk_TYPE_num` should not be in this list because it already has a special semantics defined on NULL input:
```
returns the number of elements in sk or -1 if sk is NULL.
```

Reply:
Actually, better do not say anything regarding index out of range in this new text because for each of the four functions where this can happen, its special behavior is already well defined.

Reply:
> Better add that calling these functions is discouraged on a NULL stack

I disagree with that. So I am not going to add  it.

Reply:
> And `sk_TYPE_reserve`, `sk_TYPE_push`, `sk_TYPE_unshift`, `sk_TYPE_insert`, `sk_TYPE_set`, `sk_TYPE_sort`, `sk_TYPE_is_sorted`, and `sk_TYPE_set_cmp_func` are missing from this list.

But these should NOT be in this list!

Reply:
> And `sk_TYPE_num` should not be in this list because it already has a special semantics defined on NULL input

This is not in contradiction with the statement added.

Reply:
> > And `sk_TYPE_reserve`, `sk_TYPE_push`, `sk_TYPE_unshift`, `sk_TYPE_insert`, `sk_TYPE_set`, `sk_TYPE_sort`, `sk_TYPE_is_sorted`, and `sk_TYPE_set_cmp_func` are missing from this list.
> 
> But these should NOT be in this list!

Why not? Also these can be called with a NULL stack argument, which so far is not defined.

Reply:
Calling these functions with NULL stack or invalid index is a true error.

Reply:
I agree  that that calling them with a NULL stack is a real error, but also this is so far not made clear, while - confusingly - for other functions so far this is not (considered) an error. So need to point out this weird difference.

I cannot fully agree regarding the invalid index - note that for `sk_TYPE_insert()` is is defined that
```
 If idx is out of range the new element is appended to sk.
```